seed/views/v3/properties.py

```diff
@@ -1338,11 +1342,34 @@ def update_with_building_sync(self, request, pk):
         the_file = request.data['file']
         file_type = BuildingFile.str_to_file_type(request.data.get('file_type', 'Unknown'))
         organization_id = self.get_organization(request)
-        cycle_pk = request.query_params.get('cycle_id', None)
-        org_id = self.get_organization(self.request)
+        cycle_id = request.query_params.get('cycle_id', None)
+
+        return self._update_with_building_sync(the_file, file_type, organization_id, cycle_id, pk)
+
+    def batch_update_with_building_sync(self, properties, org_id, cycle_id, progress_key):
+        """
+        Update a list of PropertyViews with a building file. Currently only supports BuildingSync.
+        """
+        progress_data = ProgressData.from_key(progress_key)
+        if not Cycle.objects.filter(pk=cycle_id):
+            logging.warning(f'Cycle {cycle_id} does not exist')
+            return progress_data.finish_with_error(f'Cycle {cycle_id} does not exist')
 
+        results = {'success': 0, 'failure': 0}
+        for property in properties:
+            formatted_time = time.strftime('%Y%m%d_%H%M%S', time.localtime(time.time()))
+            blob = ContentFile(property['xml'], name=f'at_{property["audit_template_building_id"]}_{formatted_time}.xml')
+            response = self._update_with_building_sync(blob, 1, org_id, cycle_id, property['property_view'], property['updated_at'])
+            response = json.loads(response.content)
+            results['success' if response['success'] else 'faulure'] += 1
+
+            progress_data.step('Updating Properties...')
+
+        progress_data.finish_with_success(results)
+
+    def _update_with_building_sync(self, the_file, file_type, organization_id, cycle_id, view_id, at_updated=False):
```

Comment:
this function gets called when an update from the property detail is triggered, or when a batch_update from the data page is triggered. When the batch update is triggered a get_buildings request is first made. This returns `updated` from Audit Template, it then loops through selected buildings and `_update_with_building_sync`.

The property detail only triggers the `get_building_xml` endpoint which does not return the `updated` attribute from Audit Template. Getting the `updated` would require an additional call to Audit Template.